heltour/tournament/views.py

```diff
@@ -1839,14 +1840,16 @@ def view(self, post=False):
                     for p in player_list:
                         field_name = 'av_r%d_%s' % (r.number, p.lichess_username)
                         is_available = self.request.POST.get(field_name) != 'on'
-                        if is_available != is_available_dict.get((r.id, p.id), True):
+                        
+                        if (is_available != is_available_dict.get((r.id, p.id), True) and 
+                        SeasonPlayer.objects.filter(player=p, season=self.season).first().card_color != "red"):
                             PlayerAvailability.objects.update_or_create(player=p, round=r,
                                                                         defaults={
                                                                             'is_available': is_available})
                 return redirect('by_league:by_season:edit_availability', self.league.tag,
                                 self.season.tag)
 
-            round_data = [(r, [(p, is_available_dict.get((r.id, p.id), True)) for p in player_list])
+            round_data = [(r, [(p, is_available_dict.get((r.id, p.id), True), season_player_set.get(player=p).card_color=="red") for p in player_list])
```

Comment:
get will throw an exception if this player doesn't exist, are we sure this will never error?

Reply:
Additionally here, this will issue a single query for every player.  Perhaps we should actually query for all of the red cards, setup a dict `{<player_id>: <card_color>}` and then look up in here? It would both increase safety and performance.

Reply:
Most recent commit uses dictionary and accesses the season player objects in safer way

Reply:
This is much better. Thanks!